design/one-pager-stack-status-spec.md

```diff
@@ -1,96 +1,85 @@
 # Stack UI Schema Status Spec
 
-- Owner: Steven Rathbauer (@rathpc)
+- Owner: Steven Rathbauer ([@rathpc](https://github.com/rathpc))
 - Reviewers: Crossplane Maintainers
-- Status: Draft
+- Status: Draft, revision 1.0
 
 ## Proposal
 
 [crossplaneio/crossplane#929](https://github.com/crossplaneio/crossplane/issues/929)
 
 ## Problem
 
-For a given CRD I want to be able to expose various statuses to a frontend GUI. Initially it makes
-sense to at least have the Ready state available. In addition to the value of the Ready state it
-would also be helpful to have some additional details about that state if applicable.
+For a given CRD I want to be able to expose various condition statuses to a frontend GUI. In addition to the condition
+type and status it would also be helpful to have some additional details about that type if applicable, such as message
+or lastTransitionTime.
 
-This also assumes that statuses are following a particular condition convention, similar to that of a
-[podcondition](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.16/#podcondition-v1-core)
+This should assume that condition statuses are following a particular condition convention, similar to that of a
+[podcondition]
 
 The conditions should contain at least the following keys:
 - type
 - status
-- reason
 - message
+- lastTransitionTime
 
 #### For example:
 
-If a CRD has the condition type 'Ready' defined but has a status of that type of
-_"False"_, I could show some visual indication to a user through a GUI that the CRD is **NOT Ready**.
+If a CRD has the condition type 'Ready' defined but has a status of that type of _"False"_, you could then show some
+visual indication to a user through a GUI that the CRD is **NOT Ready**.
 
-When that status changes to _"True"_ then I can adjust that visual cue to now reflect that the CRD is
-in fact **Ready**.
+When that status changes to _"True"_ then you can adjust that visual cue to now reflect that the CRD is in fact **Ready**.
 
-As of right now there is no defined spec to be able to drive status information for a given resource from a GUI perspective. It would be useful to add a key to the existing `ui-schema.yaml`
-file that can be used for this purpose.
-
-This key would contain an array of JSONPath's that point directly to conditions following the convention described above.
+As of right now there is no defined spec to be able to drive status information for a given resource from a GUI
+perspective. It would be useful to add a key to the existing `ui-schema.yaml` file that can be used for this purpose.
 
 This 1-pager will identify the key that should be added and what that structure looks like.
 
 ## Design
 
-#### Add two new keys to the `ui-schema.yaml` file:
+#### New key to be added to the `ui-schema.yaml` file:
 
-- `resourceConditionPaths` _(this is an array of JSONPaths)_
+- `printerColumns` _(this is a string or map of key/value pairs)_
```

Comment:
For code generation or usage in typed languages, there could be a benefit to keeping this variable defined as a map and perhaps using a difference variable for the string interpretation.

That said, I don't know of any doctrine that recommends avoiding variable types.  Was there an existing pattern you were pairing this with? It feels (Kubernetes) familiar, but I can't place it.

Reply:
This is adjacent to this PR which uses a similar pattern https://github.com/crossplaneio/crossplane/pull/1144

Reply:
Discussed offline and settled with https://github.com/crossplaneio/crossplane/pull/1144/files#r362861161